### ISSUE TITLE
Application gateway ingress controller with multiple hosts and ssl

### ISSUE THREAD

**Issue by balazs-pinter** [Azure/application-gateway-kubernetes-ingress#317]
**Describe the bug**
I have an application gateway ingress controller with multiple hosts and rules and TLS. But the  ingress controller cannot push the changeset into application gateway. The error I'm getting is:

`BEGIN ApplicationGateway deployment
W0626 10:24:00.484639       1 controller.go:119] unable to send CreateOrUpdate request, error [network.ApplicationGatewaysClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidRequestFormat" Message="Cannot parse the request." Details=[{"code":"DuplicateResourceName","message":"Resource /subscriptions//resourceGroups//providers/Microsoft.Network/applicationGateways/ has two child resources with the same name (sslr-dev-xxx-dev-ingress)."},{"code":"DuplicateResourceName","message":"Resource /subscriptions//resourceGroups//providers/Microsoft.Network/applicationGateways/ has two child resources with the same name (sslr-dev-xxx-dev-ingress)."}]]
I0626 10:24:00.484666       1 controller.go:120] END ApplicationGateway deployment
I0626 10:24:00.484674       1 eventqueue.go:132] Processing event failed`

**Ingress Controller details**

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    certmanager.k8s.io/cluster-issuer: letsencrypt-prod
    flux.weave.works/antecedent: dev:helmrelease/xxx-dev
    kubernetes.io/ingress.class: azure/application-gateway
    service.beta.kubernetes.io/azure-load-balancer-internal: "true"
  creationTimestamp: "2019-06-24T11:33:41Z"
  generation: 5
  labels:
    app.kubernetes.io/instance: xxx-dev
    app.kubernetes.io/managed-by: Tiller
    app.kubernetes.io/name: xxx
    helm.sh/chart: xxx-1.0.0
  name: xxx-dev-ingress
  namespace: dev
  resourceVersion: "2467557"
  selfLink: /apis/extensions/v1beta1/namespaces/dev/ingresses/xxx-dev-ingress
  uid: ea86a40a-9673-11e9-a71f-b2fef94ebab4
spec:
  rules:
  - host: app1.domain.com
    http:
      paths:
      - backend:
          serviceName: app1-service
          servicePort: 1080
  - host: app2.domain.com
    http:
      paths:
      - backend:
          serviceName: app2-service
          servicePort: 2338
  - host: app3.domain.com
    http:
      paths:
      - backend:
          serviceName: app3-service
          servicePort: 3338
  tls:
  - hosts:
    - app1.domain.com
    secretName: app1-cert
  - hosts:
    - app2.domain.com
    secretName: app2-cert
  - hosts:
    - app3.domain.com
    secretName: app3-cert
status:
  loadBalancer: {}
```

**Comment by draychev** [comment]
@balazs-pinter  Which version of AGIC are you running?
If not the latest (`0.7.0-rc2` as of this writing) - would you be so kind to upgrade?

**Comment by draychev** [comment]
@balazs-pinter I used v `0.7.0-rc2` to the Ingress definition you shared.  

I [augmented your YAML](https://gist.github.com/draychev/a6d45921202fd6e0a62ec25543c7cc80) to add Services.
Also added `appgw.ingress.kubernetes.io/ssl-redirect: "true"` annotation since from the error message you shared - it seems that your intent was to enable SSL Redirect?

I also removed `service.beta.kubernetes.io/azure-load-balancer-internal: "true"` since it semantically conflicts with  `kubernetes.io/ingress.class: azure/application-gateway`. (App Gateway will be your load balancer)

My AGIC was able to successfully post config to ARM and properly configured App Gwy SSL Redirects:
![Annotation 2019-06-26 122542](https://user-images.githubusercontent.com/49918230/60208673-b9295000-980d-11e9-8893-de58085b807e.png)

I suspect that the bug you ran into is one we already addressed a few releases ago.
